### PR TITLE
#4702: General order chat methods

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-core/issues/4702 |
| Dependencies | https://github.com/ripe-tech/ripe-core/pull/4706 |
| Decisions | Implement methods to interact with an order's general chat. |

### CODE REVIEW

File: src/js/api/order.js [hunk 1]
```diff
@@ -549,6 +547,155 @@ ripe.Ripe.prototype.logOrderP = function(number, options) {
     });
 };
 
+/**
+ * Returns the chat of an order.
```
Comment (BeeMargarida) [inline src/js/api/order.js:551]: Perhaps add more information like:
```suggestion
 * Returns the general chat of an order, not related to any state.
```

File: src/js/api/order.js [hunk 2]
```diff
@@ -549,6 +547,155 @@ ripe.Ripe.prototype.logOrderP = function(number, options) {
...
+ripe.Ripe.prototype.chatOrderP = function(number, options) {
+    return new Promise((resolve, reject) => {
+        this.chatOrder(number, options, (result, isValid, request) => {
+            isValid ? resolve(result) : reject(new ripe.RemoteError(request, null, result));
+        });
+    });
+};
+
+/**
+ * Returns the chat lines of an order.
```
Comment (BeeMargarida) [inline src/js/api/order.js:587]: Add more information about the fact that it contains the state changes lines and information about it, ordered chronologically.

**Review by BeeMargarida — APPROVED** [review]
nice :ok_hand:

File: src/js/api/order.js [hunk 3]
```diff
@@ -549,6 +547,165 @@ ripe.Ripe.prototype.logOrderP = function(number, options) {
...
+
+/**
+ * Adds a new general chat line message to a specific order.
+ *
+ * @param {Number} number The number of the order where the chat line message will be added.
+ * @param {Object} contents The contents of the chat message.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {Promise} An object containing the chat lines for the specific order.
+ */
+ripe.Ripe.prototype.chatCreateLineOrderP = function(number, contents, options) {
```
Comment (3rdvision) [inline src/js/api/order.js:701]: to follow the same format as other methods such as `createReturnWaybillOrderP`

```suggestion
ripe.Ripe.prototype.createChatLineOrderP = function(number, contents, options) {

```

File: src/js/api/order.js [hunk 4]
```diff
@@ -549,6 +547,165 @@ ripe.Ripe.prototype.logOrderP = function(number, options) {
...
+/**
+ * Adds a new general chat line message to a specific order.
+ *
+ * @param {Number} number The number of the order where the chat line message will be added.
+ * @param {Object} contents The contents of the chat message.
+ * @param {Object} options An object of options to configure the request.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.chatCreateLineOrder = function(number, contents, options, callback) {
```
Comment (3rdvision) [inline src/js/api/order.js:677]: to follow the same format as other methods 

```suggestion
ripe.Ripe.prototype.createChatLineOrder = function(number, contents, options, callback) {

```

File: src/js/api/order.js (hunk 3 above)
Comment (joao-conde) [reply src/js/api/order.js:701]: Following the same formatting as state chat: https://github.com/ripe-tech/ripe-sdk/blob/jc/4702-order-chat-methods/src/js/api/order.js#L914

File: src/js/api/order.js (hunk 4 above)
Comment (joao-conde) [reply src/js/api/order.js:677]: Following the same formatting as state chat: https://github.com/ripe-tech/ripe-sdk/blob/jc/4702-order-chat-methods/src/js/api/order.js#L914

File: src/js/api/order.js (hunk 3 above)
Comment (3rdvision) [reply src/js/api/order.js:701]: > Following the same formatting as state chat: https://github.com/ripe-tech/ripe-sdk/blob/jc/4702-order-chat-methods/src/js/api/order.js#L914

I understand that but I think we follow the same format that is in the majority of the methods to keep consistency

File: src/js/api/order.js (hunk 4 above)
Comment (3rdvision) [reply src/js/api/order.js:677]: I understand that but I think we follow the same format that is in the majority of the methods to keep consistency

File: src/js/api/order.js (hunk 3 above)
Comment (joao-conde) [reply src/js/api/order.js:701]: But it should also be a direct naming from Core, which was already approved: https://github.com/ripe-tech/ripe-core/blob/master/src/ripe_core/controllers/api/order.py#L226

Comment (3rdvision) [reply src/js/api/order.js:701]: Then I guess the effort to change isn't worth

**Review by joamag — APPROVED** [review]

**Pull request merged**